tools/autograd/derivatives.yaml

```diff
@@ -762,6 +762,15 @@
   index: non_differentiable
   result: self_t.index_copy_(dim, index, source_t)
 
+- name: index_copy(Tensor self, int dim, Tensor index, Tensor source) -> Tensor
```

Comment:
This pattern of having two version of the derivative formula (one for out-of-place and one for inplace) to avoid a perf regression looks like it might happen in a few more places. Can you add a note with a quick comment at the `index_add_()` formula, and then reference it at the `index_copy_()` formula? You can mention that the separate formula for the inplace derivative isn't required, but we have it for performance: ([from Alban's comment](https://github.com/pytorch/pytorch/pull/65993#discussion_r730008207)).


We have a convention for "Note" comments in the codebase - something like `Note [Separate Derivative Formula For Inplace]` at the first site and `See Note [Separate Derivative Formula For Inplace]` at other places that want to reference it.

Reply:
Actually, there was more discussion about this ([in the description of this PR](https://github.com/pytorch/pytorch/pull/71751#issue-1113389832)). The updated guidance is that we're okay with removing the inplace derivative formula from these ops (both `index_copy_` and `index_add_`) and taking the perf regression. There's an outstanding github issue for adding codegen to automatically recover the perf ([here](https://github.com/pytorch/pytorch/issues/71987#issue-1117494030)).

So you can ignore the above comment, and just delete both inplace formulas. Sorry about that :)

Reply:
Thanks, @bdhirsh for the comment and update. Apologies for the delay in responding, just trying to finish the opened PRs.

I've made the required changes, I hope it looks good now.